onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -249,6 +202,8 @@ Status Conv<T>::UpdateState(OpKernelContext* context, bool bias_expected) const
     }
 
     if (!s_.cached_benchmark_results.contains(x_dims_cudnn)) {
+      IAllocatorUniquePtr<void> algo_search_workspace = GetScratchBuffer<void>(AlgoSearchWorkspaceSize);
```

Comment:
Can keep this in the switch branch that does the EXHAUSTIVE algo search. We only need to allocate the workspace for that. It can be skipped in the other 2 scenarios. Was reported by an internal team a while back.